evennia/contrib/worldloader/command.py

```diff
@@ -0,0 +1,281 @@
+# import csv tables
+#------------------------------------------------------------
+class CmdImportCsv(MuxCommand):
+    """
+    Usage:
+      @importcsv
+      
+      <modelname> must be in world_settings.CSV_DATA.
+      If <modelname> is empty, it will import all csv files in world_settings.CSV_DATA.
+    """
+    key = "@importcsv"
+    locks = "perm(Builders)"
+    help_cateogory = "Builders"
+    arg_regex = r"\s.*?|$"
+
+    def func(self):
+        "Implement the command"
+    
+        caller = self.caller
+        
+        # count the number of files loaded
+        count = 0
+        
+        # get app_name
+        app_name = world_settings.WORLD_DATA_APP
+        
+        # get model_name, can specify the model name in args
+        # if no args is given, load all models in settings.WORLD_DATA
+        models = self.args
+        if models:
+            models = [arg.strip() for arg in models.split(',')]
+        else:
+            models = world_settings.WORLD_DATA
+
+        # import models one by one
+        for model_name in models:
+            # can only import models in world_settings.WORLD_DATA
+            if not model_name in world_settings.WORLD_DATA:
+                caller.msg("%s is not in world_settings.WORLD_DATA, cannot import." % model_name)
+                continue
+
+            # make file name
+            file_name = os.path.join(world_settings.CSV_DATA_PATH, model_name + ".csv")
+            
+            # import data
+            try:
+                import_csv(file_name, app_name, model_name)
+                caller.msg("%s imported." % model_name)
+                count += 1
+            except Exception, e:
+                print e
+                continue
+
+        caller.msg("total %d files imported." % count)
+
+
+#------------------------------------------------------------
+# set object's info_db and info_key
+#------------------------------------------------------------
+class CmdSetDataInfo(MuxCommand):
+    """
+    Usage:
+    @datainfo <obj>[=<key>]
+    
+    This will set the data key to an object.
+    @datainfo <obj> will show the data key of the object.
+    """
+    key = "@datainfo"
+    locks = "perm(Builders)"
+    help_cateogory = "Building"
+    
+    def func(self):
+        """
+        Implement the command
+        """
+        caller = self.caller
+        if not self.args:
+            string = "Usage: @datainfo <obj>[=<key>]"
+            caller.msg(string)
+            return
+        
+        if not self.rhs:
+            if self.args == self.lhs:
+                # no "="
+                obj_name = self.args
+                obj = caller.search(obj_name, location=caller.location)
+                if not obj:
+                    caller.msg("Sorry, can not find %s." % obj_name)
+                elif obj.db.info_db and obj.db.info_key:
+                    caller.msg("%s's datainfo is %s" % (obj_name, obj.db.info_db + "." + obj.db.info_key))
```

Comment:
With ```app``` and ```model```, the program can find the key quickly. But not all object in game has these values, such as Limbo. If I want to add an exit to it, I need to add a key to Limbo manually. Then the program can find it and add an exit to it.